api/server/router/system/system_routes.go

```diff
@@ -100,6 +100,12 @@ func (s *systemRouter) getInfo(ctx context.Context, w http.ResponseWriter, r *ht
 			// Containerd field introduced in API v1.46.
 			info.Containerd = nil
 		}
+		if versions.LessThan(version, "1.47") {
+			// Field is omitted in API 1.48 and up, but should still be included
+			// in older versions, even if no values are set.
+			info.RegistryConfig.AllowNondistributableArtifactsCIDRs = []*registry.NetIPNet{}
+			info.RegistryConfig.AllowNondistributableArtifactsHostnames = []string{}
```

Comment:
Your description says these would be `null` -- won't this send the empty list instead?  (I'm guessing `nil` gets wiped out by `omitempty`?  does the empty list not?)

Reply:
Yeah, this became more involved than I wanted. I really _thought_ Go made a distinction between `nil` and `empty slice`, but it ONLY does so if the slice itself is a pointer, but not for a slice; for a slice, _BOTH_ `nil` and `[]string{}` are considered "empty" and will be omitted, whereas for a pointer slice, `&[]string{}` is considered _non-empty_, and only `nil` is considered empty.

Really confusing! https://go.dev/play/p/wCYmrMljBzO

```go
package main

import (
	"encoding/json"
	"fmt"
)

type Foo struct {
	FieldPointerSlice          *[]string
	FieldSlice                 []string
	FieldPointerSliceOmitEmpty *[]string `json:",omitempty"`
	FieldSliceOmitEmpty        []string  `json:",omitempty"`
}

func main() {
	allEmpty := Foo{}
	out, _ := json.Marshal(&allEmpty)
	fmt.Println(string(out))

	allEmptySlice := Foo{
		FieldPointerSlice:          &[]string{},
		FieldSlice:                 []string{},
		FieldPointerSliceOmitEmpty: &[]string{},
		FieldSliceOmitEmpty:        []string{},
	}
	out, _ = json.Marshal(&allEmptySlice)
	fmt.Println(string(out))
}
```


I was hoping for the `FieldSliceOmitEmpty` that setting it to an empty slice would do the trick (for old API versions), but it doesn't. hence why I had to come up with that custom Marshal func